### ISSUE TITLE
allow specifying more known_hosts files

### ISSUE THREAD

**Issue by anarcat** [DaveDavenport/rofi#928]
## Version

Version: 1.5.1

## Configuration

Output of `rofi -help` (in a [gist](https://gist.github.com/))

https://gist.github.com/anarcat/432f8cd59e5b706474690d2a897e53f2

## Launch Command

The commandline used to launch **rofi**

    rofi -modi ssh

## Steps to reproduce

* start rofi
* type a hostname stored in a different known_hosts file

## What behaviour you see

* - only the hosts found in ~/.ssh/known_hosts files are found

## What behaviour you expect to see

* - files mentioned in a `UserKnownHostsFile` file should be parsed

I use different pools of `known_hosts` files for different hosts. For example, I have the following for Debian hosts:

```
Host *.debian.org
     UserKnownHostsFile ~/.ssh/known_hosts ~/.ssh/known_hosts.debian.org
```

This allows me to sync the `known_hosts` file from a remote without clobbering my "dynamic" `known_hosts` file. Unfortunately, the nice list of human-readable hosts in `known_hosts.debian.org
` doesn't get parsed by rofi, and so a bunch of hosts are missing.

It would be great if there would be a way to specify extra `known_hosts` file for rofi to parse, dynamically. Ideally, those would be found from the `.ssh/config` file, but maybe we don't want to re-implement a ssh config file parser in rofi - just being able to add more files to the search path would be sufficient.

Thanks for your great software! :)

**Comment by DaveDavenport** [comment]
Sounds like a useful addition.

**Comment by DaveDavenport** [comment]
Can you test?

**Comment by anarcat** [comment]
> Can you test?

Sure. I downloaded the latest git and built from source. When running, however, it doesn't find my `known_hosts.debian.org` entry. Here's how I reproduce:

```
mv ~/.ssh{,.bak} # clear out ssh config
mkdir ~/.ssh
cat > ~/.ssh/config <<EOF
Host *.debian.org
     UserKnownHostsFile ~/.ssh/known_hosts ~/.ssh/known_hosts.debian.org
EOF
cat > ~/.ssh/known_hosts.debian.org <<EOF
paradis.debian.org,paradis,eu.ssh.debian.org,eu.ssh,na.ssh.debian.org,na.ssh,ssh.debian.org,ssh,people.debian.org,people,5.153.231.30,2001:41c8:1000:21::21:30 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC2nP2WzQ8hkZDiEq5+QSGFhTB7SYfRa7/IG0gMOIoCDXH8b3sfsBUUEL8ZSFaWgNKskUnpgg/fqAhvnDLOPskr3OGXRfrCR68fCqn5H1zBrHB1kpdjPW9ezUe1xoY3hGp6LITANHWpZie1wBjYAzaBO70hNAo4JMCQvZXDsQdyws2DRSuYtiAoG/ZY0+t2VZh3MJLcofv1wNo43M+aHJR2xWmQSE7cWjMlcw/QOmsWJBv1+Kb/nK2Q7buX/byvY8ySu5ydATnyEinzbutZXc/t/FioOtWMeqh6NlD3aPGIpUTmf8ow+c1QwZWOC3T2GWyTD5KVmAZSm77lWkpYFcd1 root@paradis
paradis.debian.org,paradis,eu.ssh.debian.org,eu.ssh,na.ssh.debian.org,na.ssh,ssh.debian.org,ssh,people.debian.org,people,5.153.231.30,2001:41c8:1000:21::21:30 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIBsUneE642qr/kzKwJcKl9Cgog/kgCqRLZwZs4J7RRt8 root@paradis
EOF
```

Then I start rofi with `rofi -show ssh -disable-history` (otherwise it finds some of my older hosts) and nothing shows up. This also fails with this config:

```
Host *.debian.org
     UserKnownHostsFile ~/.ssh/known_hosts.debian.org
```

It *does* work with an expanded `~` however:

```
Host *.debian.org
     UserKnownHostsFile /home/anarcat/.ssh/known_hosts.debian.org
```

... but only if there's only one entry on that line.

So, TL;DR: almost there: it should expand tilde (`~`) and allow for multiple entries.

Thanks!

**Comment by DaveDavenport** [comment]
Aah forgot about adding the expand code on  that one :smile:

**Comment by anarcat** [comment]
ah, and it seems it doesn't like it if there are *multiple* `UserKnownHostsFile` entries in the file - it would pick the first one only.

**Comment by DaveDavenport** [comment]
It should, you should see the entries if you enable the Dialog.Ssh debug output.

**Comment by DaveDavenport** [comment]
pushed update.. I should not try to do things to quickly without testing.... So another quick fix without testing. :face_with_head_bandage:

**Comment by anarcat** [comment]
awesome, this seems to work now, thanks!

incidentally, this is basically the only way ssh completion works here if we ignore history, unless you disable hashing in your ~/.ssh/known_hosts ;)

**Comment by DaveDavenport** [comment]
little todo about that I think. I cannot reverse the hash :tongue:

**Comment by DaveDavenport** [comment]
can be closed?

**Comment by anarcat** [comment]
> little todo about that I think. I cannot reverse the hash :tongue:

i sure hope so! :)

> can be closed?

done, thanks!
